src/mode/mode_handler.ts

```diff
@@ -6,16 +6,25 @@ import {Mode, ModeName} from './mode';
 import CommandMode from './mode_command';
 import InsertMode from './mode_insert';
 import VisualMode from './mode_visual';
+import {Mappings} from '../mapping/main';
+import * as util from '../util';
+import {CommandLineMode} from '../cmd_line/main';
 
 export default class ModeHandler {
     private modes : Mode[];
     private statusBarItem : vscode.StatusBarItem;
+    mappings : Mappings;
 
     constructor() {
+        this.mappings = new Mappings();
+                
         this.modes = [
             new CommandMode(),
             new InsertMode(),
             new VisualMode(),
+            new CommandLineMode(this.mappings),
+            // new OperatorPendingMode();
```

Comment:
I think we need a way to allow transitioning from one mode to the next, as requested by the current mode (different to switching modes explicitly at user's request, I think). For example, after a command line action, we'll have to switch back to normal mode. Same for operator pending, I believe. I don't think this transitioning is possible with the current ModeHandler.